### PR TITLE
Do not penalize subdomains if sp is present but not equal to reject

### PR DESCRIPTION
**Pull request by jsf9k**
According to [RFC7489](https://tools.ietf.org/html/rfc7489#section-6.3),
> "sp" will be ignored for DMARC records published on subdomains of Organizational Domains
due to the effect of the DMARC policy discovery mechanism

Therefore we have chosen not to penalize subdomains for `sp!=reject` when considering whether subdomains satisfy a DMARC subdomain policy of reject.  This issue was originally brought to light in NCATS JIRA ticket CYHYDEV-761.

See also cisagov/trustymail#116 and jsf9k/cyhy-reports#32.

### CODE REVIEW

File: report/generate_trustymail_report.py [hunk 1]
```diff
@@ -397,6 +397,7 @@ def __score_domain(self, domain):
         score['num_dmarc_failures'] = num_dmarc_failures
 
         if domain['live']:
+            score['is_base_domain'] = domain['is_base_domain']
```
Comment (dav3r) [inline report/generate_trustymail_report.py:400]: Why did you add `is_base_domain` to the `score`, but then not use it?

File: report/generate_trustymail_report.py [hunk 2]
```diff
@@ -438,7 +439,15 @@ def __score_domain(self, domain):
...
-            if (
+            # According to RFC7489, "'sp' will be ignored for DMARC
+            # records published on subdomains of Organizational
+            # Domains due to the effect of the DMARC policy discovery
+            # mechanism."  Therefore we have chosen not to penalize
+            # for sp!=reject when considering subdomains.
+            #
+            # See here for more details:
+            # https://tools.ietf.org/html/rfc7489#section-6.3
+            if not domain['is_base_domain'] or (
```
Comment (dav3r) [inline report/generate_trustymail_report.py]: Doesn't this logic allow a subdomain with invalid DMARC to get `valid_dmarc_subdomain_policy_reject = True`?  Maybe that's ok?  I'm not quite sure.

**Review by dav3r — APPROVED** [review]
I think this looks good, but please take a look at my comments.

Props for remembering to also make this same change to the Cyber Exposure Scorecard! 🎉

File: report/generate_trustymail_report.py (hunk 1 above)
Comment (jsf9k) [reply report/generate_trustymail_report.py:400]: I was going to use it in the mustache file but ended up not needing it.  I didn't see any harm in leaving it in there.  Do you?

Comment (dav3r) [reply report/generate_trustymail_report.py:400]: Nope, just was wondering why- that is good by me.

File: report/generate_trustymail_report.py (hunk 2 above)
Comment (jsf9k) [reply report/generate_trustymail_report.py]: I think you're right.  This should be:
```python
if score['valid_dmarc'] and (not domain['is_base_domain'] or domain['dmarc_subdomain_policy'] == 'reject'):
```

Does that look correct to you?

**Comment by jsf9k** [comment]
FWIW, my intent is to do a full test scan with this code in my staging environment next week.

File: report/generate_trustymail_report.py (hunk 2 above)
Comment (dav3r) [reply report/generate_trustymail_report.py]: Yes, I think that is better.  Don't forget to make the corresponding changes in the Cyber Exposure scorecard logic.

Comment (jsf9k) [reply report/generate_trustymail_report.py]: Good catch!
:eagle: :eye:

Comment (jsf9k) [reply report/generate_trustymail_report.py]: Fixed in 6ad8b62.

**Review by dav3r — APPROVED** [review]
Looks great now.  Strong work as always @jsf9k 💪 💼
